### ISSUE TITLE
when i upload image it alerts true and did not upload

### ISSUE THREAD

**Issue by gorsargsyan** [vova07/yii2-imperavi-widget#54]

**Comment by ElisDN** [comment]
@gorsargsyan Did you specify 'image-upload' in your controller actions() method?

**Comment by kiwinewt** [comment]
I'm getting the same issue, with that action set in my controller

**Comment by kiwinewt** [comment]
FYI...
My view:
```
    <?= Redactor::widget([
        'selector' => '#post-content',
        'settings' => [
            'lang' => 'en',
            'minHeight' => 200,
            'imageManagerJson' => Url::to(['/admin/post/images-get']),
            'imageUpload' => Url::to(['/admin/post/image-upload']),
            'plugins' => [
                'imagemanager'
            ],
            'imageUploadErrorCallback' => new \yii\web\JsExpression('function(json){ alert(json.error); }')
        ]
    ]); ?>
```
Controller:
```
    public function actions()
    {
        return [
            'images-get' => [
                'class' => 'vova07\imperavi\actions\GetAction',
                'url' => Yii::$app->getUrlManager()->getBaseUrl().'/images/', // Directory URL address, where files are stored.
                'path' => Yii::$app->basePath.'/web/images/', // Or absolute path to directory where files are stored.
                'type' => GetAction::TYPE_IMAGES,
            ],
            'image-upload' => [
                'class' => 'app\modules\admin\actions\UploadAction',
                'url' => Yii::$app->getUrlManager()->getBaseUrl().'/images/', // Directory URL address, where files are stored.
                'path' => Yii::$app->basePath.'/web/images/', // Or absolute path to directory where files are stored.
                'validatorOptions' => [
                    'resizeImage' => false,
                ],
            ],
        ];
    }
```
And custom action:
```
    public function run()
    {
        if (Yii::$app->request->isPost) {
            $file = UploadedFile::getInstanceByName($this->uploadParam);
            $model = new DynamicModel(compact('file'));
            if(!$this->validatorOptions['resizeImage']) {
                $model->addRule('file', $this->_validator, $this->validatorOptions)->validate();
            }

            if ($model->hasErrors()) {
                $result = [
                    'error' => $model->getFirstError('file')
                ];
            } else {
                if ($this->unique === true && $model->file->extension) {
                    $model->file->name = uniqid() . '.' . $model->file->extension;
                }
                if ($model->file->saveAs($this->path . $model->file->name)) {
                    if(isset($this->validatorOptions['resizeImage']) && $this->validatorOptions['resizeImage']) {
                        list($oWidth, $oHeight) = getimagesize($this->path . $model->file->name);
                        $width = $oWidth;
                        $height = $oHeight;
                        if (0 < $oWidth && 0 < $oHeight) {
                            if (($oWidth > $oHeight) && $oWidth > $this->validatorOptions['maxWidth']) {
                                $width = $this->validatorOptions['maxWidth'];
                                $height = round(($oHeight / $oWidth) * $width);
                            } elseif ($oHeight > $this->validatorOptions['maxHeight']) {
                                $height = $this->validatorOptions['maxHeight'];
                                $width = round(($oWidth / $oHeight) * $height);
                            }
                        }
                        Image::thumbnail($this->path . $model->file->name, $width, $height)->save($this->path . $model->file->name, ['quality' => 80]);
                    }
                    $result = ['filelink' => $this->url . $model->file->name];
                    if ($this->uploadOnlyImage !== true) {
                        $result['filename'] = $model->file->name;
                    }
                } else {
                    $result = [
                        'error' => Yii::t('vova07/imperavi', 'ERROR_CAN_NOT_UPLOAD_FILE')
                    ];
                }
            }
            Yii::$app->response->format = Response::FORMAT_JSON;

            return $result;
        } else {
            throw new BadRequestHttpException('Only POST is allowed');
        }
    }
```

**Comment by kiwinewt** [comment]
If I remove this section from my custom action it works fine:

            if(!$this->validatorOptions['resizeImage']) {
                $model->addRule('file', $this->_validator, $this->validatorOptions)->validate();
            }

**Comment by vova07** [comment]
The `true` is alerted when some error occur.
After https://github.com/vova07/yii2-imperavi-widget/pull/108 will be merged, the new message will alert instead.
I suppose that this issue can be closed as soon as #108 will be merged.

**Comment by vova07** [comment]
Fixed in https://github.com/vova07/yii2-imperavi-widget/pull/108
